lib/jsonwp-proxy/proxy.js

```diff
@@ -19,15 +20,34 @@ class JWProxy {
       port: 4444,
       base: '/wd/hub',
       sessionId: null,
-      timeout: REQUEST_TIMEOUT,
+      timeout: DEFAULT_REQUEST_TIMEOUT,
     }, opts);
     this.scheme = this.scheme.toLowerCase();
+    this._activeRequests = [];
   }
 
   // abstract the call behind a member function
   // so that we can mock it in tests
   async request (...args) {
-    return await request(...args);
+    const currentRequest = request(...args);
+    this._activeRequests.push(r);
+    return await currentRequest.finally(() => {
+                    _.remove(this._activeRequests, (r) => {
+                      r === currentRequest;
+                    });
+                  });
+  }
+
+  hasActiveRequests () {
+    return this._activeRequests.length > 0;
```

Comment:
It might complain about unnecessary `!!`. It would also be perfectly fine to return `this._activeRequests.length` since `0` is falsy and anything else is truthy.